app/controllers/api/v2/hostgroups_controller.rb

```diff
@@ -73,6 +73,9 @@ def create
         @hostgroup.suggest_default_pxe_loader if params[:hostgroup] && params[:hostgroup][:pxe_loader].nil?
 
         process_response @hostgroup.save
+      rescue ActiveRecord::InvalidForeignKey => error
+        e = error.message.scan(/\(([^)]+)\)/)
+        not_found _("%{resource} with id %{id} not found") % {:resource => e[0][0], :id => e[1][0]}
```

Comment:
Could the message be a bit more descriptive? e.g. mentioning that it's in fact an associated resource that couldn't be found. This would generate "Subnet with id 3 not found" in case I'm creating hostgroup." Perhaps "Associated resource %{resource} with id %{id} not found" would be clearer.